### PR TITLE
Handle SUMA registry file

### PR DESCRIPTION
**Pull request by jesusbv**
- Add SUSE registry FQDN to the SUMA if not present

### CODE REVIEW

File: lib/cloudregister/registerutils.py [hunk 1]
```diff
@@ -2027,6 +2035,41 @@ def has_network_access_by_ip_address(server_ip):
     return True
 
 
+# ----------------------------------------------------------------------------
+def is_suma_instance():
+    """Return a list containing the SUMA product if present."""
+    products_list = glob.glob('/etc/products.d/*.prod')
+    suma_prod = [
+        product for product in products_list if 'suse' in product.lower() and
+        'manager' in product.lower()
```
Comment (schaefi) [inline lib/cloudregister/registerutils.py:2044]: Let's hope that suse has no product that contains 'suse' and 'manager' in its name. Call me pedantic but couldn't we match on a clear name ? I'm not too deep into suma but I can imagine it has a unique identifier ?

File: lib/cloudregister/registerutils.py [hunk 2]
```diff
@@ -2394,3 +2437,90 @@ def __set_registries_conf_docker(private_registry_fqdn):
...
+        return True
+
+    suma_yaml_content = {}
+    suma_yaml_content, failed = get_suma_registry_content()
+    if failed:
+        return False
+
+    if not suma_yaml_content or suma_yaml_content == {'registry': ''}:
+        # empty content or no other content than registry key and empty value
+        os.unlink(SUMA_REGISTRY_CONF_PATH)
```
Comment (schaefi) [inline lib/cloudregister/registerutils.py:2493]: Would removing that file harm SUMA ?

File: lib/cloudregister/registerutils.py [hunk 3]
```diff
@@ -2394,3 +2437,90 @@ def __set_registries_conf_docker(private_registry_fqdn):
...
+            # registry value does not contain registry nor susecloud.net
+            return True
+
+    if suma_yaml_content.pop('registry', False):
+        if suma_yaml_content:
+            logging.info('Updating %s after removing registry entry')
+            return __write_suma_conf(suma_yaml_content)
+
+    # no other content than the removed registry key
+    os.unlink(SUMA_REGISTRY_CONF_PATH)
```
Comment (schaefi) [inline lib/cloudregister/registerutils.py:2521]: Same here, would removing the file harm the functionality of SUMA itself ?

**Review by schaefi — COMMENTED** [review]
Thanks Jesus :+1: Implementation looks good but I'm unsure if there is maybe a more stable way to detect the product and I don't know if we can remove /etc/uyuni/uyuni-tools.yaml without harming SUMA itself ? Maybe something to check prior merge ?

File: lib/cloudregister/registerutils.py (hunk 1 above)
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:2044]: we can check the exact match which is `SUSE-Manager-Server.prod` as of today

File: lib/cloudregister/registerutils.py (hunk 2 above)
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:2493]: When I asked that I ws told if the file is not present SUMA would use the builtin defaults, however, the image
should contain that file and it should not get removed

Will edit the code to reflect that

Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: if the file should be removed and the file should be baked in with the image + containing the registry FQDN
maybe should only add it if not present and let it be, meaning not removing that entry when deregistering 

Thoughts ?

File: lib/cloudregister/registerutils.py (hunk 1 above)
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: Updated the check for SUMA instance

File: lib/cloudregister/registerutils.py (hunk 2 above)
Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: I have removed the SUMA file cleanup, we set the value if not present as the value should be present and `cloud-regionsrv-client` does not touch that file once value is set

File: lib/cloudregister/registerutils.py [hunk 4]
```diff
@@ -2027,6 +2034,42 @@ def has_network_access_by_ip_address(server_ip):
...
+            suma_reg_config_data = yaml.safe_load(suma_reg_config)
+        return suma_reg_config_data, False
+    except IOError as error:
+        logging.info(str(error))
+        action = 'open'
+    except yaml.YAMLError:
+        action = 'parse'
+
+    logging.info('Could not %s %s' % (action, SUMA_REGISTRY_CONF_PATH))
+    failed = __mv_file_backup(SUMA_REGISTRY_CONF_PATH)
```
Comment (jesusbv) [inline lib/cloudregister/registerutils.py:2069]: I'm not 100 % sure we should move the file to a backup, that would make `cloud-regionsrv-client` the owner of that file and IMO it should be owned by SUMA

that would mean we log that file could not be opened or parsed and no further action

File: lib/cloudregister/registerutils.py (hunk 2 above)
Comment (schaefi) [reply lib/cloudregister/registerutils.py]: perfect, so that's clear then. Thanks

Comment (schaefi) [reply lib/cloudregister/registerutils.py]: Another point that we need to clarify. When we write that file, what needs to be done that SUMA reads it ? Maybe we need to send a signal or restart a service, or at least document the needed steps regarding suma after the registration. @jesusbv you mentioned a meeting with the SUMA people, can you ask that question there ? Thanks

File: lib/cloudregister/registerutils.py (hunk 4 above)
Comment (schaefi) [reply lib/cloudregister/registerutils.py:2069]: I agree, the backup strategy in this case might be unexpected. Just the logging is imho enough

File: lib/cloudregister/registerutils.py (hunk 1 above)
Comment (schaefi) [reply lib/cloudregister/registerutils.py]: I like @rjschwei feedback regarding this matching

Comment (schaefi) [reply lib/cloudregister/registerutils.py]: my gut feeling is, I would just check for the exact .prod file to identify the product and that's it

Comment (schaefi) [reply lib/cloudregister/registerutils.py]: given the suse madness about names I might be wrong here, but in that case matching for 'suse' and 'manager' could also fail when they decide to name it suse-mgr or s.u.s.e.-mgr or maybe some other crazy name to make live of developers hard :-))

Comment (jesusbv) [reply lib/cloudregister/registerutils.py]: the [code](href="https://confluence.suse.com/display/publiccloud/Roadmap">https:/) has changed ? 
can you see the new code, @schaefi ?

**Review by rjschwei — APPROVED** [review]

**Comment by jesusbv** [comment]
will squash the commits and merge

**Pull request merged**